t/escape.t

```diff
@@ -73,6 +74,25 @@ is
     'it should recognize character groups'
     ;
 
+is
+    uri_escape ('abcd-', '\w'),
+    '%61%62%63%64-',
+    'it should allow character class escapes'
+    ;
```

Comment:
nice improvement (sorry I introduced this mess by not testing more possibilities)
Playing with multiple variants (using this PR changes) here is  one more regex compile error - string `{}-^` as well as `{}\-^`.

To work it has to contain `\\-`

Reply:
It seems acceptable to me that it fails to compile.

Reply:
`{}-^` only fails because `}` is a higher codepoint than `^`. I've added additional tests demonstrating that characters like ^ are allowed in the character classes, and checking for errors when given an invalid range like that.